### PR TITLE
Feature/list and create applications

### PR DESCRIPTION
**Pull request by GrahamMWillis**
https://eaflood.atlassian.net/browse/SDDSIP-224?atlOrigin=eyJpIjoiYjk5ODhhNGFhZmE5NDdiNDk4OTExODZjZjQ2YTRhMTkiLCJwIjoiaiJ9

(1) Add session cache management coupled to the session cookie. 
(2) Decorate the request object with cache utility functions
(3) Add an authorization scheme to the server based on the cookie/cache functionality (1) 
(4) Add IDM login and registration pages in place of the not-ready-yet DEFRA identity system.
(5) Add functionality to interface with the API to create new applications and reference numbers.
(6) Redesign applications list page using the nun junks templates.

### CODE REVIEW

File: packages/web-service/src/pages/applications/applications.js [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+  const { userId } = journeyData
+  const applications = await APIRequests.APPLICATION.findByUser(userId)
+  console.log(JSON.stringify(applications))
+  Intl.DateTimeFormat('en-GB', { year: 'numeric', month: 'long', day: 'numeric' }).format(new Date())
+  return {
+    applications: applications
+      .map(a => ({
+        ...a,
+        lastSaved: Intl.DateTimeFormat('en-GB', { year: 'numeric', month: 'long', day: 'numeric' })
+          .format(new Date(a.updatedAt))
```
Comment (bensagar-ea) [inline packages/web-service/src/pages/applications/applications.js:16]: Is there a reason for using this format? The data standard is to use ISO 8601 for formatting dates

File: packages/web-service/assets/sass/_general.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+div.hidden-area {
```
Comment (bensagar-ea) [inline packages/web-service/assets/sass/_general.scss:1]: Is this necessary? Can't it just be `hidden`?

**Review by bensagar-ea — CHANGES_REQUESTED** [review]
Couple of things I think need changing, plus there's a few files with things like no newline at the end and blank lines at the beginning that could be tidied up with linting.

File: packages/web-service/src/pages/applications/applications.js (hunk 1 above)
Comment (GrahamMWillis) [reply packages/web-service/src/pages/applications/applications.js:16]: It's the display format (e.g. 28 March 2022) specified by Dave on the prototype. I could make this a filter function. Internally I will use Date Objects. I think the Intl functions are the recommended way of formatting Javascript dates now that  Moment is being obsoleted.

File: packages/web-service/assets/sass/_general.scss (hunk 2 above)
Comment (GrahamMWillis) [reply packages/web-service/assets/sass/_general.scss:1]: I did not know about <div hidden> I will change

Comment (GrahamMWillis) [reply packages/web-service/assets/sass/_general.scss:1]: `<div hidden>`

File: packages/web-service/src/pages/applications/applications.js (hunk 1 above)
Comment (bensagar-ea) [reply packages/web-service/src/pages/applications/applications.js:16]: I think that's my mistake, I hadn't properly processed the fact the object was just for display purposes.

File: packages/web-service/src/pages/applications/applications.js [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
+import { APIRequests } from '../../services/api-requests.js'
 
-const getData = async _request => []
-const setData = async _request => {}
+export const getData = async request => {
+  const journeyData = await request.cache().getData()
+  const { userId } = journeyData
+  const applications = await APIRequests.APPLICATION.findByUser(userId)
+  console.log(JSON.stringify(applications))
+  Intl.DateTimeFormat('en-GB', { year: 'numeric', month: 'long', day: 'numeric' }).format(new Date())
```
[the remark this answers is not in the document]
Comment (GrahamMWillis) [reply packages/web-service/src/pages/applications/applications.js]: Removed

**Pull request merged**